src/scripts/main.js

```diff
@@ -1 +1,55 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+body.addEventListener('contextmenu', (e) => {
+  e.preventDefault();
+});
+
+const promise1 = new Promise((resolve, reject) => {
+  body.addEventListener('mousedown', ev => {
+    if (ev.button === 0) {
+      resolve();
+    };
+  });
+
+  setTimeout(() => reject(new Error()), 3000);
+});
+
+const promise2 = new Promise((resolve) => {
+  body.addEventListener('mousedown', ev => {
+    if (ev.button === 0 || ev.button === 2) {
+      resolve();
+    };
+  });
+});
+
+const promise3 = new Promise((resolve) => {
+  body.addEventListener('mousedown', ev1 => {
+    body.addEventListener('mousedown', ev2 => {
+      if ((ev1.button === 0 && ev2.button === 2)
+      || (ev1.button === 2 && ev2.button === 0)) {
+        resolve();
+      };
+    });
+  });
```

Comment:
It's better avoid nested listeners. Create two boolean variables `rightClick` `leftClick` and set them true value if it was right or left click on body. Then just check this 2 variables and if theese variables are right resolve 3d promise